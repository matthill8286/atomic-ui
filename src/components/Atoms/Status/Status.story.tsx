import { radios } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import React from 'react'
import { Status, StatusProps, StatusType } from './Status'
import { Icon } from '@/components/Atoms/Icon'
import { OtherBookmarkOutlined } from '@matthill8286/atomic-icon-library'

const statusTypes: { [key: number]: StatusType } = { 0: 'white', 1: 'black', 2: 'primary' }

const knobs = (): StatusProps => {
  return {
    type: radios('Type', statusTypes, statusTypes[2]),
  }
}

storiesOf('Design System/Atoms/Status', module)
  .add('Default', () => {
    return <Status {...knobs()}>1</Status>
  })
  .add('With Icon', () => {
    return (
      <Icon color="grey4" height="md">
        <OtherBookmarkOutlined />
        <Status {...knobs()} withinIcon>
          1
        </Status>
      </Icon>
    )
  })
