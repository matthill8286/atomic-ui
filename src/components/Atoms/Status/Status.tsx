import { css, styled } from '@/styles/styled'

export type StatusType = 'white' | 'black' | 'primary'

export interface StatusProps {
  type?: StatusType
  withinIcon?: boolean
}

export const Status = styled.div<StatusProps>`
  ${({ withinIcon, theme }) =>
    withinIcon &&
    css`
      position: absolute;
      margin-left: ${theme.spacing.base.xs};
      margin-top: -${theme.spacing.base.xs};
      z-index: 90;
    `};
  width: ${({ theme }) => theme.spacing.base.sm};
  height: ${({ theme }) => theme.spacing.base.sm};
  background: ${({ theme, type = 'primary' }) => theme.color[type]};
  font-family: ${({ theme }) => theme.font.family.default};
  font-size: ${({ theme }) => theme.font.size.xxs};
  color: ${({ theme, type = 'primary' }) =>
    type === 'white' ? theme.color.grey6 : theme.color.white};
  border-radius: 100%;
  text-align: center;
  display: flex;
  justify-content: center;
  align-items: center;
`
