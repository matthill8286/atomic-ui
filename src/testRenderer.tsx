import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { saiyanTheme } from '@/styles/sc-vars-saiyan'
import { Theme } from '@/types'

export function renderWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>,
  theme = saiyanTheme
) {
  return renderer.create(<ThemeProvider theme={theme}>{tree}</ThemeProvider>).toJSON()
}

export function renderWithThemeAndRouter(tree: React.ReactElement, theme: Theme = saiyanTheme) {
  return renderer
    .create(
      <ThemeProvider theme={theme}>
        <Router>{tree}</Router>
      </ThemeProvider>
    )
    .toJSON()
}

export function renderWithRouter(tree: React.ReactElement) {
  return renderer.create(<Router>{tree}</Router>).toJSON()
}

export function mountWithTheme<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(<ThemeProvider theme={saiyanTheme}>{tree}</ThemeProvider>)
}

export function mountWithThemeAndRouter<C extends React.Component, P = C['props'], S = C['state']>(
  tree: React.ReactElement<P>
): ReactWrapper<P, S, C> {
  return mount(
    <ThemeProvider theme={saiyanTheme}>
      <Router>{tree}</Router>
    </ThemeProvider>
  )
}
