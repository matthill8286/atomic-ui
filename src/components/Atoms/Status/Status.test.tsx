import React from 'react'
import { renderWithTheme } from '@/testRenderer'
import { Status } from './Status'

describe('Status', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Status type={'primary'}>1</Status>)
    expect(tree).toMatchSnapshot()
  })
})
