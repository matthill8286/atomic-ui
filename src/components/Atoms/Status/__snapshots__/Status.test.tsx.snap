// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status renders correctly 1`] = `
.c0 {
  width: 16px;
  height: 16px;
  background: #00C8DC;
  font-family: OswaldLight,Arial,sans-serif;
  font-size: 12px;
  color: #FFFFFF;
  border-radius: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
  type="primary"
>
  1
</div>
`;
